apollo-federation/src/sources/connect/validation/connect/selection/variables.rs

```diff
@@ -234,29 +235,29 @@ impl NamespaceResolver for ArgsResolver<'_> {
         expression: GraphQLString,
         schema: &SchemaInfo,
     ) -> Result<(), Message> {
-        let Some(root) = get_root(reference) else {
-            return Ok(()); // Not something we can type check this way TODO: delete all of this when Shape is available
-        };
+        for (root, sub_trie) in reference.selection.iter() {
+            let field_type = self
+                .field
+                .arguments
+                .iter()
+                .find(|arg| arg.name == root)
+                .ok_or_else(|| Message {
+                    code: Code::UndefinedArgument,
+                    message: format!(
+                        "`{object}` does not have an argument named `{root}`",
+                        object = self.field.name,
+                    ),
+                    locations: reference
+                        .selection
+                        .key_ranges(root)
+                        .flat_map(|range| expression.line_col_for_subslice(range, schema))
+                        .collect(),
+                })
+                .map(|field| field.ty.clone())?;
```

Comment:
maybe `map` before the `ok_or_else`? Know the action before defining the error case

Reply:
That code structure was already here, and I'm mostly just reindenting it, but I will try putting `map` first.

Reply:
https://github.com/apollographql/router/pull/7419/commits/b62dfa0d61a4df34dd342adac0de8ad43e1031bf